web/packages/teleport/src/Discover/Shared/ConnectionDiagnostic/useConnectionDiagnostic.ts

```diff
@@ -20,55 +20,89 @@ import useAttempt from 'shared/hooks/useAttemptNext';
 import useTeleport from 'teleport/useTeleport';
 import { useDiscover } from 'teleport/Discover/useDiscover';
 import { DiscoverEventStatus } from 'teleport/services/userEvent';
+import { getDatabaseProtocol } from 'teleport/Discover/Database/resources';
 
 import type {
   ConnectionDiagnostic,
   ConnectionDiagnosticRequest,
 } from 'teleport/services/agents';
-import type { AgentStepProps } from '../../types';
+import type { MfaAuthnResponse } from 'teleport/services/mfa';
+import type { Database } from 'teleport/Discover/Database/resources';
 
-export function useConnectionDiagnostic(props: AgentStepProps) {
+export function useConnectionDiagnostic() {
   const ctx = useTeleport();
 
-  const { attempt, run } = useAttempt('');
+  const { attempt, setAttempt, handleError } = useAttempt('');
   const [diagnosis, setDiagnosis] = useState<ConnectionDiagnostic>();
   const [ranDiagnosis, setRanDiagnosis] = useState(false);
-  const { emitErrorEvent, emitEvent } = useDiscover();
+  const { emitErrorEvent, emitEvent, prevStep, nextStep, resourceState } =
+    useDiscover();
 
   const access = ctx.storeUser.getConnectionDiagnosticAccess();
   const canTestConnection = access.create && access.edit && access.read;
 
-  function runConnectionDiagnostic(req: ConnectionDiagnosticRequest) {
+  const [showMfaDialog, setShowMfaDialog] = useState(false);
+
+  // runConnectionDiagnostic will initially make a call to check if
+  // resource target requires MFA authentication. After this initial
+  // check depending on if user successfully authenticated or not (
+  // determined by the presence of the token field), will make a call
+  // to test connection.
```

Comment:
i found your explanation confusing, so i cleared it up like this (i try to follow golang convention commenting so i don't have to think too hard when switching):

```
  // runConnectionDiagnostic depending on the value of `mfaAuthnResponse` does the following:
  //   1) If param `mfaAuthnResponse` is undefined or null, it will check if MFA is required.
  //      - If MFA is required, it sets a flag that indicates a users
  //        MFA credentials are required, and skips the request to test connection.
  //      - If MFA is NOT required, it makes the request to test connection.
  //   2) If param `mfaAuthnResponse` is defined, it skips checking if MFA is required,
  //      and makes the request to test connection.
```